### PR TITLE
refactor Subriemannian metric

### PR DESCRIPTION
**Pull request by SaitejaUtpala**

### CODE REVIEW

**Comment by SaitejaUtpala** [comment]
@ninamiolane  not sure why codecov is failing... I added every test(you can verify) I think -0.01% is floating error.

File: tests/tests_geomstats/test_sub_riemannian_metric.py [hunk 1]
```diff
@@ -17,75 +19,126 @@ def cometric_matrix(self, base_point=None):
...
+    def test_hamiltonian(self, metric, cotangent_vec, base_point, expected):
         state = gs.array([base_point, cotangent_vec])
-
-        result = self.example_metric.hamiltonian(state)
-        expected = 1.5
+        result = metric.hamiltonian(state)
         self.assertAllClose(result, expected)
 
     @geomstats.tests.autograd_tf_and_torch_only
-    def test_symp_grad(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_sub_riemannian_metric.py:48]: ok

File: tests/tests_geomstats/test_sub_riemannian_metric.py [hunk 2]
```diff
@@ -17,75 +19,126 @@ def cometric_matrix(self, base_point=None):
...
-
-        result = self.example_metric.iterate(step(step_size), n_steps)(test_state)[-10]
-        expected = gs.array([[1.22, 1.33, 1.44], [2.0, 3.0, 4.0]])
+    def test_iterate(self, metric, test_state, n_steps, step_size, expected):
+        step = metric.symp_euler
+        result = metric.iterate(step(step_size), n_steps)(test_state)[-10]
         self.assertAllClose(result, expected)
 
     @geomstats.tests.autograd_tf_and_torch_only
-    def test_symp_flow(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_sub_riemannian_metric.py:75]: ok

**Review by ninamiolane — APPROVED** [review]
Fantastic, thank you very much!

**Pull request merged**